test/crd.go

```diff
@@ -48,6 +48,15 @@ type ResourceNames struct {
 	TrafficTarget string
 	Domain        string
 	Image         string
+	Namespace     string
+}
+
+// GetNamespaceOrDefault returns the default namespace or the namespace declared in the construction of "ResourceNames"
+func (n ResourceNames) GetNamespaceOrDefault() string {
```

Comment:
This name is confusing given that "default" is a special namespace name in Kubernetes and I don't think that is what you are referring to here but rather the client default.

Also "Get" in golang is fairly uncommon ( See https://golang.org/doc/effective_go.html#Getters ). I would just simplify this down to `Namespace()`, or if you want to be verbose `NamespaceOrClientDefault()`.

Reply:
Oops, updated to `NamespaceOrClientDefault()`